src/main/java/org/opensearch/sdk/ExtensionsRunner.java

```diff
@@ -108,7 +136,21 @@ InitializeExtensionsResponse handleExtensionInitRequest(InitializeExtensionsRequ
         logger.info("Registering Extension Request received from OpenSearch");
         InitializeExtensionsResponse initializeExtensionsResponse = new InitializeExtensionsResponse(extensionSettings.getExtensionName());
         opensearchNode = extensionInitRequest.getSourceNode();
+        // Fetch the unique ID
+        for (DiscoveryExtension de : extensionInitRequest.getExtensions()) {
+            if (de.getName().equals(extensionSettings.getExtensionName())) {
+                setUniqueId(de.getId());
+                break;
+            }
+        }
+        if (getUniqueId() == null) {
+            throw new IllegalArgumentException(
+                "Extension " + extensionSettings.getExtensionName() + " does not match any requested extension."
+            );
+        }
         setOpensearchNode(opensearchNode);
+        extensionTransportService.connectToNode(opensearchNode);
+        sendRegisterApiRequest(extensionTransportService);
```

Comment:
We are trying to send RegisterApi before responding to the init extension request. 
Wouldn't this mess up the workflow on ExtensionsOrchestrator? 
Can we call the API after the registration is successful ? 

I am worried if the response is not sent cleanly and something fails, we might have to build fail-safe code on ExtensionOrchestrator. Instead, I think its simpler to have the API call to register them after the extension registration is successful. WDYT?

Reply:
Excellent question, and I went back and forth on this.  I can easily switch the register request into a `finally` block to execute after the response.  I'll update again to do this but I think the question about how to handle failures needs further discussion and would welcome @owaiskazi19 and @joshpalis input on this.  A similar question should be asked about registering the named writ(e)ables.

One underlying thought I had:  the extension isn't properly initialized until both the REST API and the named writables are registered.  So we do have to handle failures of those methods in any case.

You saw in my initial draft PR that I simply replaced the original response with the API, eliminating the sequencing issue, but that doesn't allow for the API to be updated later during operation.

Reply:
I'm going to create a new issue to raise the exception handling question in the previous comment.  For now I've reordered the responses to align with the design diagrams on #64 and elsewhere.